docs/clickhouse-ingress.md

```diff
@@ -0,0 +1,166 @@
+- [Step 4: Configure Antrea Flow Aggregator](#step-4-configure-antrea-flow-aggregator)
+<!-- /toc -->
+
+## Overview
+
+In this guide, we'll walk you through the process of setting up Kubernetes Ingress
+with ClickHouse as the backend service and how to connect your Flow Aggregator to
+a ClickHouse server. Kubernetes Ingress is a powerful way to manage external
+access to services within your Kubernetes cluster, and ClickHouse is a popular
+open-source columnar database management system. By using Ingress, you can easily
+expose ClickHouse to external clients and manage routing and load balancing
+efficiently.
+
+## Prerequisites
+
+Before proceeding with the setup, ensure that you have the following prerequisites:
+
+1. A Kubernetes cluster up and running.
+2. kubectl command-line tool installed and configured to access your Kubernetes cluster.
+3. ClickHouse server deployed within your Kubernetes cluster.
+
+## Step 1: Install and Configure Ingress Controller
+
+The first step is to install an Ingress controller in your Kubernetes cluster.
+There are various Ingress controllers available, such as Nginx Ingress Controller,
+Traefik, etc. Choose one that suits your requirements and install it following
+the documentation provided by the controller's maintainers.
+
+Here we use Nginx Ingress Controller as an example:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.8.1/deploy/static/provider/cloud/deploy.yaml
+```
+
+## Step 2: Expose ClickHouse Service via Ingress
+
+Assuming you have already deployed ClickHouse as a Kubernetes service, you
+need to expose it using an Ingress resource. Create an Ingress manifest file
+(e.g., clickhouse-ingress.yaml) with the following content:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: clickhouse-ingress
+  namespace: flow-visibility   # Replace with your desired namespace
+spec:
+  ingressClassName: nginx   # Replace with your desired namespace or omit it for using default ingress class
+  rules:
+    - host: clickhouse.example.com   # Replace with your desired domain/subdomain
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: clickhouse-clickhouse    # Replace with your ClickHouse service name
+                port:
+                  number: 8123             # Replace with the ClickHouse service port
+  tls: # optional
+    - hosts:
+        - clickhouse.example.com       # Replace with your desired domain/subdomain
+      secretName: clickhouse-tls-secret   # Replace with the name of your TLS secret
+```
+
+Save the file and deploy the Ingress resource using kubectl:
+
+```bash
+kubectl apply -f clickhouse-ingress.yaml
+```
+
+Ensure that the Ingress resource is created successfully:
+
+```bash
+kubectl get ingress clickhouse-ingress
+```
+
+## Step 3: Create TLS Secret (Optional)
```

Comment:
Do you think we can also mention that user can use cert-manager like here https://github.com/antrea-io/antrea/blob/2302cfeae5e1e54e96e93c0130cdfbac097c471f/docs/securing-control-plane.md?plain=1#L115

Reply:
I've added cert-manager section and update the doc. Thanks.